cli/internal/run/dry_run.go

```diff
@@ -44,8 +45,11 @@ func DryRun(
 			taskSummary.Framework = runsummary.MissingFrameworkLabel
 		}
 
+		// This mutex is not _really_ required, since we are using Concurrency: 1 as an execution
+		// option, but we add it here to match the shape of RealRuns execFunc.
```

Comment:
Not a strong feeling, but this feels unnecessary. I think a comment above `Concurrency: 1` would be just fine.